api/context.go

```diff
@@ -135,6 +140,20 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if sessionId != "" {
+		c.TeamUrl = c.Session.TeamURL
+		c.Path = r.URL.Path
+	} else if h.isApi || h.isTeamIndependent {
+		c.TeamUrl = c.SiteUrl
+		c.Path = r.URL.Path
+	} else {
+		splitURL := strings.Split(r.URL.Path, "/")
+		c.TeamUrl = protocol + "://" + r.Host + "/" + splitURL[1]
+		c.Path = "/" + strings.Join(splitURL[2:], "/")
+	}
+
+	w.Header().Set(model.HEADER_TEAM_URL, c.TeamUrl)
```

Comment:
This was originally added when I was keeping consistancy between the two url modes. I kept it in case we ever do want to support both. Also fixing the client to always use this header would make things cleaner instead of parsing window.location.